src/sparse_tabular.jl

```diff
@@ -199,9 +199,9 @@ function reward_s_a(mdp::Union{MDP, POMDP})
     action_space = actions(mdp)
     reward_S_A = fill(-Inf, (length(state_space), length(action_space))) # set reward for all actions to -Inf unless they are in actions(mdp, s)
     for s in state_space
-        if isterminal(mdp, s)
-            reward_S_A[stateindex(mdp, s), :] .= 0.0
-        else
+        # if isterminal(mdp, s)
```

Comment:
same as for transition matrix, do we want to check here or do we expect the check to happen in `reward(...)`?

Reply:
IIRC we didn't want an extra overhead in thinking about this from the user side.

Reply:
Ok I guess that works. Should we then document somewhere that the user doesn't need to worry about checking for terminal states in `reward` and `transition`? Right now it seems like the code for this is duplicated.
Also, re-introducing these lines I commented out will make some tests fail for the TMaze because it allows "escaping" from the terminal state. I guess that's a bug that needs to be taken care of on the TMaze side of things though

Reply:
This is very important! Since no actions will be taken from a terminal state (https://juliapomdp.github.io/POMDPs.jl/stable/basic_properties/#Terminal-States-1), a complete and valid problem implementation might have a `transition` function that errors when it is called on a terminal state. Thus, we need this check to avoid such errors.

> Should we then document somewhere that the user doesn't need to worry about checking for terminal states in reward and transition?

Yes, this is a good fact to know, but where should we put it? Happy to add it if there is a good place.

> I guess that's a bug that needs to be taken care of on the TMaze side of things though

Yes, thanks for finding that!

Reply:
I guess it could be added here https://juliapomdp.github.io/POMDPs.jl/stable/explicit/ ? 
@zsunberg